Migrating optimize webfonts article

rachelandrew opened the pull request:
Fixes #3332

Changes proposed in this pull request:

- Create a new article /optimize-webfonts/
- Updated fast.js to add a link to it

A couple of questions for @kaycebasques and/or @mihajlija  

1. The article uses webfonts (all one word) I have seen both webfonts and web fonts used however, it's not in the word list. Just checking what the preference is.
2. The latter part of the article was obviously written when font-display was new, I am not sure we need all of the information here as we have other articles that cover font-display (which I have linked to). 

I've already done some updating to describe current status and also added a link to the article on variable fonts which aren't mentioned in this, but let me know if you think we could drop more of this.

mihajlija commented:
It's "WebFonts" at [W3C](https://www.w3.org/Fonts/WG/) so I suggest we stick with that. I'll add it to the word list.

mihajlija commented:
Hi @rachelandrew, I started reviewing this and trying to figure out how it can better complement the existing content. One thing that keeps bothering me is this post's length, it's about four times as long as anything else on /fast. It's a very much WebFundamentals-style post, while web.dev strives to have shorter, more focused posts bundled into learning paths. 

I suggest we chunk it up, here's how I would re-arrange it: 

# Optimize webfonts
- Reduce font size (this would include compression and `@font-face`)
- Optimize font loading and rendering (this would include a shorter intro to font-display and preloading since we already have posts for that, and Font Loading API and caching)
- https://web.dev/avoid-invisible-text/
- https://web.dev/preload-optional-fonts/

I'd use parts of the checklist from the post and relevant Lighthouse audits for the conclusion/summary in revised posts. 

Do you have time to make these changes? If you are busy working on something else atm, I could probably squeeze this in this week :)

rachelandrew commented:
@mihajlija I'm happy to carry on with it, it was really why I posted my progress as I wasn't sure how much to carve it up!

mihajlija commented:
@rachelandrew Awesome! Keep an eye out for duplicated information, such as the font-display table for browser behavior (https://web.dev/avoid-invisible-text/ has one that's up to date). 

Other minor nits: 
- Formatting of WOFF 2(.0) is inconsistent.
- Midway through the post "webfont" becomes "Webfont".
- There are relative links to WebFundamentals website that won't work on web.dev.

rachelandrew commented:
In the commits today I have:

- Split the article into two as suggested above.
- Changed everything to use WebFont / WebFonts as per the W3C.
- Made mentions of WOFF 2.0 consistent as per the W3C.
- Removed content which exists in more recent articles.
- Fixed links.
- Also fixed links in /preload-critical-assets which linked to the WebFu version of this.
- Edited copy to fit more with web.dev style.
- Updated fast.js.

cc: @mihajlija

kaycebasques commented on src/site/content/en/fast/reduce-webfont-size/index.md:
```diff
@@ -0,0 +1,360 @@
+
+The `@font-face` CSS at-rule allows you to define the location of a particular font resource, its
+style characteristics, and the Unicode codepoints for which it should be used. A combination of such
+`@font-face declarations can be used to construct a "font family," which the browser will use to
```
```suggestion
`@font-face` declarations can be used to construct a "font family," which the browser will use to
```

kaycebasques commented on src/site/content/en/fast/optimize-webfont-loading/index.md:
```diff
@@ -0,0 +1,213 @@
+Understanding these periods means you can use `font-display` to decide how your
+font should render depending on whether or when it was downloaded.
+
+To work with the `font-display` property, add it your `@font-face` rules:
```
```suggestion
To work with the `font-display` property, add it to your `@font-face` rules:
```

kaycebasques commented on src/site/content/en/fast/optimize-webfont-loading/index.md:
````diff
@@ -0,0 +1,213 @@
+}
+```
+
+`font-display` currently supports the following range of values:
+`auto | block | swap | fallback | optional`.
````
```suggestion
`font-display` currently supports the following range of values:

* `auto`
* `block`
* `swap`
* `fallback`
* `optional`
```

kaycebasques commented on src/site/content/en/fast/optimize-webfont-loading/index.md:
```diff
@@ -0,0 +1,213 @@
+Used together, `<link rel="preload">` and the CSS `font-display` give you a great deal of control over font loading and rendering,
+without adding in much overhead.
+But if you need additional customizations,
+and are willing to incur with the overhead introduced by running JavaScript, there is another option.
```
```suggestion
and are willing to incur the overhead introduced by running JavaScript, there is another option.
```

kaycebasques commented on src/site/content/en/fast/optimize-webfont-loading/index.md:
````diff
@@ -0,0 +1,213 @@
+```
+
+Further, because you can check the font status
+(via the [check()](https://www.w3.org/TR/css-font-loading/#font-face-set-check)) method
````
```suggestion
(via the [`check()`](https://www.w3.org/TR/css-font-loading/#font-face-set-check)) method
```

rachelandrew commented:
- PR for WebFundamentals https://github.com/google/WebFundamentals/pull/8861
- I added WebFont to the word list in my last commit.
